ipn/local.go

```diff
@@ -1124,9 +1127,7 @@ func (b *LocalBackend) SetPrefs(newp *Prefs) {
 // painstakingly constructing it in twelvety other places.
 func (b *LocalBackend) doSetHostinfoFilterServices(hi *tailcfg.Hostinfo) {
 	hi2 := *hi
-	if b.shieldsAreUp() {
-		// No local services are available, since ShieldsUp will block
-		// them all.
+	if !b.shouldUploadServices() {
```

Comment:
Not in this commit, but I wonder whether we should stop *collecting* the services in this case, rather than stop *reporting* them.

Reply:
I think collection only happens on startup right now so opt-in/opt-out changes wouldn't take effect automatically, but yea we could just not collect unless needed (and could similarly turn on/off collection here?)

Reply:
I see. Collection only on startup makes services a bit less useful, since (hopefully) tailscale has long uptimes. Might be worth filing an issue to follow up on this.

Reply:
Yeah good point

Reply:
It collects every few seconds before & after this change, not just at start-up.

Changing that is a bigger change.